### PR TITLE
HDDS-8340. [Recon] Show the number of deleted containers in UI

### PR DESCRIPTION
**Pull request by smitajoshi12**
## What changes were proposed in this pull request?
This information will be displayed on the overview page of recon which gets populated with the information provided by the ClusterStateEndpoint

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-8340

Please replace this section with the link to the Apache JIRA)

## How was this patch tested?
Manually
image

### CODE REVIEW

**Comment by kerneltime** [comment]
@devmadhuu @xBis7 can you'll please take a look?

**Comment by xBis7** [comment]
IMHO, displaying the number of missing containers would be much more useful than the number of deleted containers. Does it make more sense to see how many of the existing containers have gone missing or are unhealthy? Or how many data you didn't need and eventually deleted?

**Comment by devmadhuu** [comment]
> IMHO, displaying the number of missing containers would be much more useful than the number of deleted containers. Does it make more sense to see how many of the existing containers have gone missing or are unhealthy? Or how many data you didn't need and eventually deleted?

@xBis7 Thanks for your review, I think Recon displays the missing containers and unhealthy containers under "Containers" tab. This PR is mainly focussed to show count on Overview page.

**Comment by xBis7** [comment]
> I think Recon displays the missing containers and unhealthy containers under "Containers" tab. This PR is mainly focussed to show count on Overview page.

@devmadhuu I know and also there is the number of containers on each tab(missing, under-replicated, ...). Why would the user want to see the number of deleted containers?

**Comment by devmadhuu** [comment]
> > I think Recon displays the missing containers and unhealthy containers under "Containers" tab. This PR is mainly focussed to show count on Overview page.
> 
> @devmadhuu I know and also there is the number of containers next to each tab header (missing, under-replicated, ...). Why would the user want to see the number of deleted containers?

@xBis7 as far as i know , this count of deleted containers was done based on multiple issues being faced while debugging customer issues, so needed this info to correlate with information in rocks db. Mainly targeted for admin users not for direct customers.

**Comment by xBis7** [comment]
> this count of deleted containers was done based on multiple issues being faced while debugging customer issues, so needed this info to correlate with information in rocks db. Mainly targeted for admin users not for direct customers.

@devmadhuu Thanks for the info. Based on that, this PR looks good. 

CI failure seems unrelated.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx [hunk 1]
```diff
@@ -227,6 +231,9 @@ export class Overview extends React.Component<Record<string, object>, IOverviewS
           <Col xs={24} sm={18} md={12} lg={12} xl={6}>
             <OverviewCard loading={loading} title='Keys' data={keys.toString()} icon='file-text'/>
           </Col>
+          <Col xs={24} sm={18} md={12} lg={12} xl={6}>
+            <OverviewCard loading={loading} title='Deleted Containers' data={deletedContainers !== undefined && deletedContainers.toString()} icon='delete' />
```
Comment (xBis7) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:235]: This check seems redundant `deletedContainers !== undefined`, unless there is a chance that the server response won't contain it.

Comment (xBis7) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:235]: Can you do a manual test with docker, without the `pnpm` tool?

Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:235]: @xBis7
I checked with docker also working fine with undefined condition.
undefined is additional check if deletedContainers will fail in API Response.

Comment (dombizita) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:235]: I am also not sure why the `deletedContainers !== undefined` check is needed, non of the above `OverviewCard`s contain it.
why can't we assume that the API response will always return this information? the backend part of this have been in done in #4440 and I don't see anything there that would require this check, right @ArafatKhan2198? 
btw I tried in a docker cluster and it was working for me.

Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx]: @dombizita 
Removed undefined condition in latest commit.

**Review by dombizita — COMMENTED** [review]
thanks for working on this @smitajoshi12, it looks good to me!

**Review by dombizita — APPROVED** [review]

**Review by smengcl — APPROVED** [review]

**Comment by smengcl** [comment]
Thanks @smitajoshi12 for the addition. Thanks @xBis7 @devmadhuu @dombizita for the reviews.

**Pull request merged**